### DISCUSSION TITLE
Would be cool if it could replace EDscout

### DISCUSSION THREAD

**Discussion by packet1** [iaincollins/icarus#1] [Ideas & Suggestions]
I love what EDScout provides and what love to see that capability incorporated here!

**Comment by ayaanrauf** [comment]
Agreed, maybe it could also show the distance and have a route plotter or something like that in the future?

**Reply by iaincollins** [reply]
> Agreed, maybe it could also show the distance and have a route plotter or something like that in the future?

We can definitely do distance and will look at adding that in.

I would love to a route planner, although I think that is probably where I have to draw the line as would need to use a third party service to handle it - similar to the way ICARUS Terminal uses [EDSM](https://www.edsm.net/) for system maps - as the entire galaxy is several GB (compressed!) and that's a lot of data bundle each copy of the app, especially as the current app is only about 0.02 GB and it would pose practical problems.

I would rather not maintain a live service in this case (I just don't want the hassle) so it would depend on something like [spansh.co.uk](https://spansh.co.uk/) being okay with ICARUS using an API hosted there to do that. That is something I would like to explore with maintainers of services like that though.

We are really lucky with EDSM, as it has a really simple open API. EDSM also has a route plotter, and maybe they would be willing to allow usage of it from an external app.

The other thing that is a bit of kick in the pants here is that the game data flow is one way and you can't set routes in game from a third party tool, which is a shame. It would be really nice to be able to set routes and target objects in a system from third party tools. If you could do those it would also mean you could just give the game a system name and have it do the route plotting logic, which would simplify things enormously, but I'm still very grateful we have an API for the game at all (something that is all too often neglected I think).

**Reply by ayaanrauf** [reply]
Alright, thank you! Didn't know the drawbacks of these features before, now I understand.

**Reply by fudoido** [reply]
"can't set routes in game from a third party tool, which is a shame"
you cant inject it into the game directly, but by using the keyboard controls assigned to it, it can be done. edcopilot does this for a while now.
it depends big time in voice attack, although can be used standalone (clicking menus and writing stuff, but with VA is so much better and easier), you say "plot route for xyz19", and it will open the map, search the system (writing the actual name in the search bar), select it and set the route, then closes map. all without you pressing a key, just by saying it. it can plan a route for trades or even road2riches kinda plan, using spansh i believe.
What i am loving in your tool is the graphic art on it, really want to use them as browser plugin and make use of my 2 tablets, might not even use it as active tools, but want it to make my cockpit stand out :p unfortunately so far i wasnt able to use browser with it, tablets browsers keep trying to load but nothing, and firewall exception was created for icarus.

**Comment by iaincollins** [comment]
> I love what EDScout provides and what love to see that capability incorporated here!

I like the approach EDScout takes too, and definitely plan to add system type, details and estimated scan value (even before I'd seen EDScout) to the Navigation list view and/or right hand inspector panel when a body is selected.

I've actually been thinking of adding a toggle to that view to display additional info in the list view, should there be more info that easily fits on one line and that's something I'll likely introduce - especially to call out interesting or valuable locations!

**Comment by iaincollins** [comment]
As a step towards this, the Navigation List view now displays icons highlighting interesting features of bodies ([more details in the v0.5.0 release notes](https://github.com/iaincollins/icarus/releases/tag/v0.5.0)).

What it doesn't do yet is submit data to EDDN - so if exploring in the back you would still want to use a program like EDDI that submits data to EDDN/EDSM to have results of scans appear in ICARUS Terminal - and it doesn't yet include an estimated value for bodies, but obviously terraformable bodies, earth-likes, water world and high metal content worlds are more valuable.

It doesn't have indicators for gas giants or ammonia worlds, they might get similar treatment in future.

![v0-5-0](https://user-images.githubusercontent.com/595695/154546607-4fe39fd4-bea0-4ec9-b021-74d88b24d85c.png)

**Comment by packet1** [comment]
The important part of EDScouts -

-  show what is discovered or not when you plot your route. 
-  More rows of data on the screen (go for scrolling long plotted routes to see what is discovered or not) .  
- Fast scrolling on tablet
- Gravity shown
- Value of body 
- The behavior of it - how it refreshes and purges systems visited on your route.   What am I kidding, I just want a clone of it in here!
